readme/set-current-user.md

```diff
@@ -0,0 +1,92 @@
+[Back to Main README](https://github.com/simplabs/ember-simple-auth/blob/master/README.md)
+
+## How to Set currentUser Data
+
+Although there are various ways to load currentUser data, this is the canonical
+way to do so. Your authentication route will need to send the id of the user
+account. In this example, it is sent as the user_id.
+
+### CurrentUser Service
+
+In order for the app to keep the user data, we will store it in its own service:
+
+```js
+// app/services/session-account.js
+import Ember from 'ember';
+
+const { inject: { service }, isEmpty, RSVP } = Ember;
+
+export default Ember.Service.extend({
+  session: service('session'),
+  store: service(),
+
+  account: null,
+
+  loadCurrentUser() {
+    return new RSVP.Promise((resolve, reject) => {
+      const userId = this.get('session.data.authenticated.user_id');
+      if (!isEmpty(userId)) {
+        return this.get('store').find('user', userId).then((user) => {
+          this.set('user', user);
+          resolve();
+        }, reject);
+      } else {
+        resolve();
+      }
+    });
+  }
+});
+```
+
+In this example, we grab the user id and then make an ember-data request to the
+API. After a successful request, it sets the account easily access anywhere by
+injecting this service.  Note: If no id is sent, then the user will not be
+set and this implementation will log you out.
+
+### Calling the function to set the current user
+
+The best place for us to make this call is on the application route:
+
+```js
+import Ember from 'ember';
+import ApplicationRouteMixin from 'ember-simple-auth/mixins/application-route-mixin';
+
+const { service } = Ember.inject;
+
+export default Ember.Route.extend(ApplicationRouteMixin, {
+  sessionAccount: service(),
+
+  beforeModel() {
+    return this._loadCurrentUser();
+  },
+
+  sessionAuthenticated() {
+    this._super(...arguments);
+    this._loadCurrentUser().catch(() => this.get('session').invalidate());
+  },
+
+  _loadCurrentUser() {
+    return this.get('sessionAccount').loadCurrentUser();
+  }
+});
+```
+
+This sets uses the service to set the when the application route loads or when
+a session is authenticated.  In this instance, if you can't load the current
+user, then it will not log you in.
```

Comment:
This should also explain why the current user needs to be loaded at 2 different places in the code - one if the application already starts up authenticated, one if the session is authenticated at a later point in time.